source/beamline/BioXAS/BioXASSSRLMonochromatorRegionControl.cpp

```diff
@@ -170,148 +170,238 @@ AMControl::FailureExplanation BioXASSSRLMonochromatorRegionControl::move(double
 
-		connect( bragg_, SIGNAL(movingChanged(bool)), this, SIGNAL(movingChanged(bool)) );
+		if (bragg_)
+			controlSetup(bragg_);
 	}
 }
 
 void BioXASSSRLMonochromatorRegionControl::setBraggAtCrystalChangePositionStatusControl(AMControl *inPosition)
 {
-	if (!braggAtCrystalChangePositionStatus_ && braggAtCrystalChangePositionStatus_ != inPosition) {
+	if (braggAtCrystalChangePositionStatus_ != inPosition) {
+
+		if (braggAtCrystalChangePositionStatus_) {
+			controlCleanup(braggAtCrystalChangePositionStatus_);
+			braggAtCrystalChangePositionStatus_ = 0;
+		}
+
 		braggAtCrystalChangePositionStatus_ = inPosition;
-		addChildControl(braggAtCrystalChangePositionStatus_);
 
-		connect( braggAtCrystalChangePositionStatus_, SIGNAL(valueChanged(double)), this, SIGNAL(braggAtCrystalChangePositionStatusChanged(double)) );
-		connect( braggAtCrystalChangePositionStatus_, SIGNAL(movingChanged(bool)), this, SIGNAL(movingChanged(bool)) );
+		if (braggAtCrystalChangePositionStatus_) {
+			controlSetup(braggAtCrystalChangePositionStatus_);
+			connect( braggAtCrystalChangePositionStatus_, SIGNAL(valueChanged(double)), this, SIGNAL(braggAtCrystalChangePositionStatusChanged(double)) );
+		}
 	}
 }
 
 void BioXASSSRLMonochromatorRegionControl::setCrystalChangeControl(AMControl *crystalChange)
 {
-	if (!crystalChange_ && crystalChange_ != crystalChange) {
+	if (crystalChange_ != crystalChange) {
+
+		if (crystalChange_) {
+			controlCleanup(crystalChange_);
+			crystalChange_ = 0;
+		}
+
 		crystalChange_ = crystalChange;
-		addChildControl(crystalChange_);
 
-		connect( crystalChange_, SIGNAL(movingChanged(bool)), this, SIGNAL(movingChanged(bool)) );
+		if (crystalChange_)
+			controlSetup(crystalChange_);
 	}
 }
 
 void BioXASSSRLMonochromatorRegionControl::setCrystalChangeCWLimitStatusControl(AMControl *limitStatus)
 {
-	if (!crystalChangeCWLimitStatus_ && crystalChangeCWLimitStatus_ != limitStatus) {
+	if (crystalChangeCWLimitStatus_ != limitStatus) {
+
+		if (crystalChangeCWLimitStatus_) {
+			controlCleanup(crystalChangeCWLimitStatus_);
+			crystalChangeCWLimitStatus_ = 0;
+		}
+
 		crystalChangeCWLimitStatus_ = limitStatus;
-		addChildControl(crystalChangeCWLimitStatus_);
 
-		connect( crystalChangeCWLimitStatus_, SIGNAL(movingChanged(bool)), this, SIGNAL(movingChanged(bool)) );
+		if (crystalChangeCWLimitStatus_)
+			controlSetup(crystalChangeCWLimitStatus_);
 	}
 }
 
 void BioXASSSRLMonochromatorRegionControl::setCrystalChangeCCWLimitStatusControl(AMControl *limitStatus)
 {
-	if (!crystalChangeCCWLimitStatus_ && crystalChangeCCWLimitStatus_ != limitStatus) {
+	if (crystalChangeCCWLimitStatus_ != limitStatus) {
+
+		if (crystalChangeCCWLimitStatus_) {
+			controlCleanup(crystalChangeCCWLimitStatus_);
+			crystalChangeCCWLimitStatus_ = 0;
+		}
+
 		crystalChangeCCWLimitStatus_ = limitStatus;
-		addChildControl(crystalChangeCWLimitStatus_);
 
-		connect( crystalChangeCCWLimitStatus_, SIGNAL(movingChanged(bool)), this, SIGNAL(movingChanged(bool)) );
```

Comment:
Yup! This connect statement (and it's parallels) should never have been there, as the composite control interface handles this behavior. I may have added them at one point to test something, but that test is no longer useful/relevant.